backend/src/main/java/org/eclipse/tractusx/puris/backend/supply/logic/service/CustomerSupplyService.java

```diff
@@ -0,0 +1,181 @@
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialServiceImpl;
+import org.eclipse.tractusx.puris.backend.stock.logic.service.MaterialItemStockService;
+import org.eclipse.tractusx.puris.backend.supply.domain.model.OwnCustomerSupply;
+import org.eclipse.tractusx.puris.backend.supply.domain.model.ReportedCustomerSupply;
+import org.eclipse.tractusx.puris.backend.supply.domain.repository.ReportedCustomerSupplyRepository;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class CustomerSupplyService {
+    private final ReportedCustomerSupplyRepository repository;
+    private final OwnDeliveryService ownDeliveryService;
+    private final ReportedDeliveryService reportedDeliveryService;
+    private final OwnDemandService demandService;
+    private final MaterialItemStockService stockService;
+    private final MaterialServiceImpl materialService;
+
+    protected final Function<ReportedCustomerSupply, Boolean> validator;
+
+    public CustomerSupplyService(
+        ReportedCustomerSupplyRepository customerRepository,
+        OwnDeliveryService ownDeliveryService,
+        ReportedDeliveryService reportedDeliveryService,
+        OwnDemandService demandService,
+        MaterialItemStockService stockService,
+        MaterialServiceImpl materialService) {
+        this.repository = customerRepository;
+        this.ownDeliveryService = ownDeliveryService;
+        this.reportedDeliveryService = reportedDeliveryService;
+        this.demandService = demandService;
+        this.stockService = stockService;
+        this.materialService = materialService;
+        this.validator = this::validate;
+    }
+
+    public final List<OwnCustomerSupply> calculateCustomerDaysOfSupply(String material, String partnerBpnl, String siteBpns, int numberOfDays) {
+        List<OwnCustomerSupply> customerSupply = new ArrayList<>();
+        LocalDate localDate = LocalDate.now();
+
+        List<Double> demands = demandService.getQuantityForDays(material, partnerBpnl, siteBpns, numberOfDays);
+        log.info("demands: " + demands);
+
+        List<DeliveryQuantityDto> ownDeliveries = ownDeliveryService.getQuantityForDays(material, partnerBpnl, siteBpns, true, numberOfDays);
+        List<DeliveryQuantityDto> reportedDeliveries = reportedDeliveryService.getQuantityForDays(material, partnerBpnl, siteBpns, true, numberOfDays);
+        List<Double> deliveries = mergeDeliveries(ownDeliveries, reportedDeliveries);
+        log.info("deliveries: " + deliveries);
+
+        double stockQuantity = stockService.getInitialStockQuantity(material, partnerBpnl, siteBpns);
+        log.info("stock qty: " + stockQuantity);
+
+        for (int i = 0; i < numberOfDays; i++) {
+            Date date = Date.from(localDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+
+            if (i == numberOfDays - 1) {
+                stockQuantity += deliveries.get(i);
```

Comment:
On the last day I need to add the delivery amount before it goes to calculate, because after it calculates, it adjusts the stock with demand and delivery, but then the loop is already finished. Maybe I complicated it and there is a better way :D And then it doesnt matter if its added again, because the loop is finished anyways.

Reply:
I'll have another look into the logic. Fix the rest of the comments for now and I'll let you know if something needs to be changed here.

Reply:
alright, deal